### DISCUSSION TITLE
Unable to communicate with OSTree

### DISCUSSION THREAD

**Discussion by gh0sti** [cockpit-project/cockpit#17606] [Q&A] [answered]
### Explain what happens

1. started messing with cockpit and making sure software was up to date
2. somehow broke communication with OSTree where cockpit can't check for software updates
3. reinstalled cockpit backports 239 can't get it to update to 270 which I was able to keep it up to date for awhile, still troubleshooting that

### Version of Cockpit

239-1

### Where is the problem in Cockpit?

Software updates

### Server operating system

other

### Server operating system version

Bullseye 11 32bit

### What browsers are you using?

Chrome

**Comment by martinpitt** [comment, marked as answer]
Seems you accidentally installed `cockpit-ostree`. Uninstall it again, and all will be happy again.

**Reply by gh0sti** [reply]
when I do sudo apt remove cockpit-ostree says it cant find that package to remove.

**Reply by martinpitt** [reply]
Please check `cockpit-bridge --packages` what "updates" points to. It should be /usr/share/cockpit/packagekit, but as it complains about OStree, it must be ".../ostree" somewhere. "bullseye" is Debian, which is classic apt. OSTree does not work on this machine. Somehow you seem to have managed to install cockpit-ostree, if not through a package then through `make install` or something similar.

**Reply by gh0sti** [reply]
I was able to remove the ostree which has fixed the issue, thank you for pointing me in that direction.

**Reply by AriMcBrown** [reply]
accidentally installed? Which "Application" includes it then? I can't recall installing it manually.

This behavior started after an update.

`sudo dnf remove ostree` fixed it for me

**Reply by unlikelyusername** [reply]
Seems this is a new/different issue.   On my system (Fedora 36), I found that automated nightly updates automatically installed cockpit-ostree,  'breaking' the software update page.

dnf.log:2022-10-29T04:08:06-0400 DEBUG Installed: cockpit-ostree-1:190.1-1.fc36.noarch

removing cockpit-ostree corrected the issue, but it should not have been automatically installed.

**Reply by garrett** [reply]
While this seems to be a problem with Fedora's dependencies, I've opened an issue about how `cockpit-ostree` could handle this better, especially as not everyone will check our issue tracker (nor should they have to).

https://github.com/cockpit-project/cockpit-ostree/issues/295

**Reply by Jonadabe** [reply]
> Please check `cockpit-bridge --packages` what "updates" points to. It should be /usr/share/cockpit/packagekit, but as it complains about OStree, it must be ".../ostree" somewhere. "bullseye" is Debian, which is classic apt. OSTree does not work on this machine. Somehow you seem to have managed to install cockpit-ostree, if not through a package then through `make install` or something similar.

Hi there.
I'm on Debian and also had installed ostree. I removed it with apt command, but I can't remember if I tried to install it with make cockpit-ostree.
What I have is this:
"cockpit-bridge --packages
apps                 Applications                             /usr/share/cockpit/apps
base1                                                         /usr/share/cockpit/base1
metrics                                                       /usr/share/cockpit/metrics
navigator            Navigator                                /usr/share/cockpit/navigator
network              Networking                               /usr/share/cockpit/networkmanager
performance                                                   /usr/share/cockpit/tuned
podman               Podman containers                        /usr/share/cockpit/podman
shell                                                         /usr/share/cockpit/shell
ssh                                                           /usr/share/cockpit/ssh
static                                                        /usr/share/cockpit/static
storage              Storage                                  /usr/share/cockpit/storaged
system               Overview, Logs, Services, Terminal       /usr/share/cockpit/systemd
**updates              Software updates                         /usr/local/share/cockpit/cockpit-ostree**
users                Accounts                                 /usr/share/cockpit/users"
How can I change it to the correct value so that the updates work again in cockpit?
Thanks

**Reply by martinpitt** [reply]
Right, please remove /usr/local/share/cockpit/cockpit-ostree to get updates working again. I suppose you once did a `sudo make install` in cockpit-ostree repo?

**Reply by Jonadabe** [reply]
Thank you. It worked!
